ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1138,24 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    /* Session version and negotiated protocol version should match
+     * (This may not be the case when reusing old session)
+     */
+    if (s->version != s->session->ssl_version) {
```

Comment:
I don't understand why this check is necessary (or the setting of `s->session->ssl_verion = s->version` a few lines above). Isn't this all covered by the `ssl_choose_client_version()` call at the beginning of this function?